### PR TITLE
Query Debugger

### PR DESCRIPTION
**Pull request by ItsMeBrianD**
### Description

This PR adds a small suite of tools to make debugging pages easier, it is mostly intended for internal use, but over time should be polished up and made available to project authors

image
image
image

### Checklist

- [x] For UI or styling changes, I have added a screenshot or gif showing before & after
- [ ] I have added a [changeset](https://github.com/evidence-dev/evidence/blob/main/CONTRIBUTING.md#adding-a-changeset)
- [ ] ~~I have added to the docs where applicable~~
- [ ] ~~ have added to the [VS Code extension](https://github.com/evidence-dev/evidence-vscode) where applicable~~

### CODE REVIEW

**Comment by ItsMeBrianD** [comment]
@mcrascal where/when do we want to land this?

File: packages/ui/core-components/src/lib/atoms/inputs/dropdown/dropdownOptionStore.spec.js [hunk 1]
```diff
@@ -268,7 +268,7 @@ describe('dropdownOptionStore', () => {
 				pauseSorting();
 				addOptions(...opts);
 				await vi.advanceTimersByTimeAsync(100);
-				expect(get(options)).toStrictEqual(opts);
+				expect(get(options)).toStrictEqual(opts.reverse());
```
Comment (zachstence) [inline packages/ui/core-components/src/lib/atoms/inputs/dropdown/dropdownOptionStore.spec.js]: Why is this change necessary?

File: packages/ui/core-components/src/lib/devtools/input-debug/Diff.svelte [hunk 2]
```diff
@@ -0,0 +1,74 @@
+<script>
+	import { formatDiff } from './formatDiff.js';
+	import formatDistanceToNowStrict from 'date-fns/formatDistanceToNowStrict';
+	import formatDate from 'date-fns/format';
+	import { readable } from 'svelte/store';
+
+	// TODO: rewrite this to be a little more accurate
```
Comment (zachstence) [inline packages/ui/core-components/src/lib/devtools/input-debug/Diff.svelte:7]: More accurate how? We should remove this comment or add more context

File: packages/lib/sdk/src/utils/svelte/inputs.js [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+		return writable({});
+	}
+	return getContext(InputStoreKey);
+};
+
+/**
+ *
+ * @returns {import("svelte/store").Readable<any>}
+ */
+export const readonlyInputContext = () => {
```
Comment (zachstence) [inline packages/lib/sdk/src/utils/svelte/inputs.js:68]:
```suggestion
export const getReadonlyInputContext = () => {
```

File: packages/ui/core-components/src/lib/devtools/query-debugger/QueryDebugger.svelte [hunk 4]
```diff
@@ -0,0 +1,124 @@
+<script context="module">
+	/** @typedef {import("@evidence-dev/sdk/usql").QueryValue} QueryValue} */
+	/**
+	 * @template T
+	 * @typedef {import("svelte/store").Writable<T>} Writable
+	 */
+	import { readonly, writable } from 'svelte/store';
+	import { batchUp } from '@evidence-dev/sdk/utils';
+
+	// TODO: is this needed?
```
Comment (zachstence) [inline packages/ui/core-components/src/lib/devtools/query-debugger/QueryDebugger.svelte:10]: ?

File: packages/ui/core-components/src/lib/devtools/query-debugger/QueryDebugger.svelte [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+	// let allQueries = derived([], (v) => v);
+
+	// /** @param {import("@evidence-dev/sdk/usql").QueryDebugValue} query */
+	// const trackQueries = (query) => {
+	// 	// TODO: Handle narrow containers
+	// 	let included = [...get(allQueries), query.proxied];
+	// 	if (!scopedOnly) included = [...ssrQueries, ...included];
+
+	// 	allQueries = derived(Array.from(new Set(included)), (v) => v);
+	// };
```
Comment (zachstence) [inline packages/ui/core-components/src/lib/devtools/query-debugger/QueryDebugger.svelte:71]: Remove these comments?

File: packages/ui/core-components/src/lib/devtools/query-debugger/sections/Verbose.svelte [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+	export let query;
+</script>
+
+<ContentBox togglable defaultOpen={false}>
+	<span slot="header">Verbose Info</span>
+	<div class="grid grid-cols-3 gap-4 w-full">
+		<div class="flex flex-col gap-2 col-span-2">
+			<span class="font-bold">Options</span>
+			<span>
+				<!-- TODO: Make sure we don't render in initialData completely -->
```
Comment (zachstence) [inline packages/ui/core-components/src/lib/devtools/query-debugger/sections/Verbose.svelte:13]: TODO?

**Review by zachstence — CHANGES_REQUESTED** [review]

**Review by zachstence — APPROVED** [review]

File: packages/ui/core-components/src/lib/atoms/accordion/AccordionItem.svelte [hunk 7]
```diff
@@ -5,14 +5,15 @@
...
+	export let compact = false;
 
 	let className = undefined;
 	export { className as class };
 </script>
 
 {#key title}
 	<BaseAccordion.Item value={title} class={className}>
-		<BaseAccordion.Trigger>
+		<BaseAccordion.Trigger class={compact ? 'py-0' : ''}>
```
Comment (ItsMeBrianD) [inline packages/ui/core-components/src/lib/atoms/accordion/AccordionItem.svelte:16]: Needs documentation

File: packages/ui/core-components/src/lib/atoms/hint/Hint.svelte [hunk 8]
```diff
@@ -10,6 +10,12 @@
 	/** @type {import('@steeze-ui/svelte-icon').IconSource} */
 	export let icon = InfoCircle;
 
+	/** @type {"sm" | "md" | "lg"}*/
+	export let maxWidth = undefined;
+
+	/** @type {"right" | "left"} */
+	export let direction = 'right';
+
```
Comment (ItsMeBrianD) [inline packages/ui/core-components/src/lib/atoms/hint/Hint.svelte:18]: documentation

**Pull request merged**